task3.c

```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+#include <string.h>
+#include <locale.h>
+
+int main()
+{
+    FILE *file = fopen("input.txt", "r");
+    if (file == NULL)
+    {
+        printf("File not found!");
+        return 1;
+    }
+
+    int frequence[255] = { 0 };
+    while(!feof(file))
+    {
+        ++frequence[fgetc(file)];
```

Comment:
```suggestion
        int currentCharCode = fgetc(file);
        if (currentCharCode >= 0)
        {
            ++frequence[currentCharCode];
        }
```
Здесь забавно получается. `fgetc` берет следующий символ из файла и переносит указатель на этот символ.
`feof` сообщит о том, что файл исчерпан, только когда указатель будет правее последнего символа в файле.
Вот и получается, что на последней итерации указатель в файле указывает на последний символ, `feof` запускает нас в тело цикла, `fgetc` возвращает следующий символ (символа нет, поэтому `-1`), и мы присваиваем по адресу `-1` в массив. То есть выходим за пределы массива.